service/remote.go

```diff
@@ -136,15 +145,15 @@ func (r *RemoteService) Call(ctx context.Context, req *protos.Request) (*protos.
 			},
 		}
 	} else {
-		res = processRemoteMessage(c, req, r)
+		processRemoteMessage(c, req, res, r)
 	}
 
 	if res.Error != nil {
 		err = errors.New(res.Error.Msg)
 	}
 
 	defer tracing.FinishSpan(c, err)
-	return res, nil
+	return &*res, nil
```

Comment:
why not simply return res instead of &*res?

Reply:
Because the return value of Call() is a pointer.
And, we want to put it back in the pool and reuse it.
Therefore, we can only return its copy value here to avoid object insecurity.

Reply:
Oh, I see!